packages/wonder-blocks-button/src/components/button-core.tsx

```diff
@@ -466,40 +478,36 @@ export const _generateStyles = (
             },
         };
     } else if (kind === "tertiary") {
+        const focusStyling = {
+            outlineStyle: "solid",
+            outlineColor: light ? theme.color.border.tertiary.inverse : color,
+            outlineWidth: theme.border.width.focused,
+            borderRadius: theme.border.radius.default,
+        };
+        const activePressedStyling = {
+            color: light ? fadedColor : activeColor,
+            textDecoration: "underline",
+            textDecorationThickness: theme.size.underline.active,
+        };
+
         newStyles = {
             default: {
                 background: "none",
                 color: light ? theme.color.text.inverse : color,
                 paddingLeft: 0,
                 paddingRight: 0,
-            },
-            hover: {
-                ":after": {
-                    content: "''",
-                    position: "absolute",
-                    height: theme.size.height.tertiaryHover,
-                    width: "100%",
-                    right: 0,
-                    bottom: 0,
-                    background: light ? theme.color.bg.tertiary.hover : color,
-                    borderRadius: theme.border.radius.tertiary,
-                },
-            },
-            focus: {
-                outlineStyle: "solid",
-                outlineColor: light
-                    ? theme.color.border.tertiary.inverse
-                    : color,
-                outlineWidth: theme.border.width.focused,
-                borderRadius: theme.border.radius.default,
-            },
-            active: {
-                color: light ? fadedColor : activeColor,
-                ":after": {
-                    height: theme.size.height.tertiaryHover,
-                    background: light ? fadedColor : activeColor,
+                [":hover:not([aria-disabled=true])" as any]: {
+                    textUnderlineOffset: theme.font.offset.default,
+                    textDecoration: "underline",
+                    textDecorationThickness: theme.size.underline.hover,
```

Comment:
**note:** Also simplified the `tertiary` hover styles to rely on underlines instead of using the `:after` hack.